### PR TITLE
Fix SqlParameter Validation for Enum Type.

### PR DESCRIPTION
**Pull request by kkurni**
Revert back the logic as close as current net framework.
Because it doesn't count if the the type is enum.

In corefx, we cannot get GetTypeCode from Type. But we can get this from the value/instance by using Convert.GetTypeCode(value).

If it is enum type, Convert.GetTypeCode will called IConvertible.GetTypeCode which implemented in Enum Type to get the underlying type.

### CODE REVIEW

**Comment by kkurni** [comment]
CC @corivera @saurabh500

**Comment by kkurni** [comment]
This will fix this issue https://github.com/dotnet/corefx/issues/1613

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlEnums.cs [hunk 1]
```diff
@@ -378,150 +378,147 @@ static internal MetaType GetMetaTypeFromValue(object value, bool inferLen = true
...
-                {
-                    return MetaXml;
-                }
+                case TypeCode.Empty:
+                    throw ADP.InvalidDataType(nameof(TypeCode.Empty));
+                case TypeCode.Object:
+                    
+                    if (dataType == typeof (System.Byte[]))
+                    {
+                        // mdac 90455 must not default to image if inferLen is false ...
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlEnums.cs]: This comment should be removed.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlEnums.cs [hunk 2]
```diff
@@ -378,150 +378,147 @@ static internal MetaType GetMetaTypeFromValue(object value, bool inferLen = true
...
+                    if (dataType == typeof (SqlBinary))
+                        return MetaVarBinary;
+                    if (dataType == typeof (SqlBoolean))
+                        return s_metaBit;
+                    if (dataType == typeof (SqlByte))
+                        return s_metaTinyInt;
+                    if (dataType == typeof (SqlBytes))
+                        return MetaVarBinary;
+                    if (dataType == typeof (SqlChars))
+                        return MetaNVarChar; // MDAC 87587
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlEnums.cs]: Remove MDAC references

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlEnums.cs [hunk 3]
```diff
@@ -378,150 +378,147 @@ static internal MetaType GetMetaTypeFromValue(object value, bool inferLen = true
...
+                        return MetaDecimal;
+                    if (dataType == typeof (SqlSingle))
+                        return s_metaReal;
+                    if (dataType == typeof (SqlXml))
+                        return MetaXml;
+                    if (dataType == typeof (SqlString))
+                    {
+                        return ((inferLen && !((SqlString) value).IsNull)
+                            ? PromoteStringType(((SqlString) value).Value)
+                            : MetaNVarChar); // MDAC 87587
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlEnums.cs]: Remove MDAC reference

**Comment by saurabh500** [comment]
@dotnet-bot Test Innerloop Windows_NT Release Build and Test please

**Comment by saurabh500** [comment]
@kkurni There are some comments which need to be cleaned up.

Otherwise LGTM.

**Pull request merged**